src/notification.ts

```diff
@@ -18,36 +18,18 @@
 
 import { fetch as crossFetch } from "cross-fetch";
 import { FetchError } from "./errors";
-
-export type protocols = "ws" | string;
-export type statuses = "connecting" | "connected" | "closing" | "closed";
-
-export type NegotiationInfo = {
-  endpoint: string;
-  procotol: protocols;
-  features: FeatureOptions;
-};
-
-export type NotificationConnectionInfo = {
-  endpoint: string;
-  protocol: protocols;
-  subprotocol: string;
-};
-
-export type BaseNotificationOptions = {
-  features?: FeatureOptions;
-  gateway?: string;
-  host?: string;
-  fetch?: typeof crossFetch;
-};
-
-export type FeatureOptions = {
-  state?: string;
-  ttl?: number;
-  rate?: number;
-  filter?: string;
-};
-
+import {
+  protocols,
+  FeatureOptions,
+  statuses,
+  BaseNotificationOptions,
+  NegotiationInfo,
+  NotificationConnectionInfo,
+} from "./interfaces";
+
+/**
+ * @internal
+ */
 export class BaseNotification {
   topic: string;
```

Comment:
All of the properties of this class should be private, and not appear in the documentation. All these values are provided by the constructor or during the negotiation with the notification endpoint, they should neither be set a posteriori not be read by a third party.